moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,449 @@
+ * on its row (i.e Schema of feature on Node of row).
+ *
+ * @param b builtin value
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_boolean_cell_exec(const builtin b,
+                            const int row_num, const string& label)
+{
+	return createLink(EXECUTION_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  mk_row_number_cell(row_num),
+	                  mk_boolean_cell(b));
+}
+
+struct mk_cell_visitor : public boost::static_visitor<Handle>
+{
+	Handle operator()(const builtin b) const
+	{
+		return mk_boolean_cell(b);
+	}
+
+	Handle operator()(const contin_t contin) const
+	{
+		return mk_real_cell(contin);
+	}
+
+	template<typename T>
+	Handle operator()(const T& t) const
+	{}
+};
+
+/**
+ * Represent a row as a list of cells.
+ *
+ * @param cells cells in a row
+ * @return
+ */
+Handle mk_row(const HandleSeq& cells)
+{
+	return createLink(cells, LIST_LINK);
+}
+
+/**
+ * Represent a row as a list of row number and cells
+ *
+ * @param cells cells in a row
+ * @param row_num row number
+ * @return
+ */
+Handle mk_numbered_row(const HandleSeq& cells, const int row_num)
+{
+	return createLink(LIST_LINK,
+	                  mk_row_number_cell(row_num),
+	                  createLink(cells, LIST_LINK));
+}
+
+/**
+ * Create a table of dependent (output) data.
+ *
+ * @param oTable output table
+ * @return
+ */
+Handle mk_output_table(const OTable& oTable)
+{
+	const string& label = oTable.get_label();
+	HandleSeq rows;
+	int row_num = 0;
+
+	for (const vertex& v : oTable)
+	{
+		Handle h = createLink(LIST_LINK,
+		                      mk_row_number_cell(++row_num),
+		                      boost::apply_visitor(mk_cell_visitor(), v));
+		rows.push_back(h);
+	}
+
+	return createLink(SIMILARITY_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  createLink(rows, SET_LINK));
+}
+
+/**
+ * Create a table of independent (input) data.
+ *
+ * @param iTable input table
+ * @param type type of the output
+ * @todo this method doesn't support a table of mixed types. It assumes that
+ * 		 all features have similar type with the output feature.
+ * @return
+ */
+Handle mk_input_table(const ITable& iTable, const type_node type)
+{
+	const vector<string> labels = iTable.get_labels();
+	HandleSeq rows;
+	int row_num = 0;
+
+	if (type == id::boolean_type)
+	{
+		for (const multi_type_seq& m : iTable)
+		{
+			HandleSeq cells;
+
+			for (builtin cell_value : m.get_seq<builtin>())
+				cells.push_back(mk_boolean_cell(cell_value));
+
+			Handle h = createLink(LIST_LINK,
+			                      mk_row_number_cell(++row_num),
+			                      createLink(cells, LIST_LINK));
+			rows.push_back(h);
+		}
+	}
+	else if (type == id::contin_type)
+	{
+		for (const multi_type_seq& m : iTable)
+		{
+			HandleSeq cells;
+
+			for (contin_t cell_value : m.get_seq<contin_t>())
+				cells.push_back(mk_real_cell(cell_value));
```

Comment:
Hmm, you could simplify here as well by having the whole `push_back` loop wrapped in a visitor, but maybe it's just too much worry. I mean this code is already quite clean as it is.